scripts/main.js

```diff
@@ -33,63 +33,103 @@ module.exports = (robot) => {
-          title = "Sprint - "+title
-          const myDate = moment(title, 'DD-MMM-YYYY').toDate()
-          if(myDate == "Invalid Date") {
-            res.reply("Title format is invalid")
-          }
-          else {
-            closeMilestoneAll(title)
-            res.reply("Milestones successfully closed")
-          }
-        }
-        else if(message.length === 6 && message[0] === "bot" && message[1] === "create" && message[2] === "milestone"){
-          var title = message[3]+" "+message[4]+" "+message[5]
-          const myDate = moment(title, 'DD-MMM-YYYY').toDate()
-          if(myDate == "Invalid Date") {
-            res.reply("Title format is invalid")
-          }
-          else {
-            createMilestoneAll(title, myDate)
-            res.reply("Milestones successfully created")
+        if(message[0] === "bot" && message[1] === "create" && message[2] === "milestone") {
+          switch(message.length) {
+            case 7: 
+              let repos = message[6]
+              repos = repos.split('/')
+              let title = "Sprint - "+message[3]+" "+message[4]+" "+message[5]
+              let myDate = moment(title, 'DD-MMM-YYYY').toDate()
+              if(myDate == "Invalid Date") {
+                res.reply("Title format is invalid")
+              }
+              else {
+                createMilestone(title, myDate, repos[0], repos[1])
+                res.reply("Milestone successfully created at 'https://github.com/"+repos[0]+"/"+repos[1]+"/milestones")
+              }
+              break;
+            case 6:
+              title = "Sprint - "+message[3]+" "+message[4]+" "+message[5]
+              myDate = moment(title, 'DD-MMM-YYYY').toDate()
+              if(myDate == "Invalid Date") {
+                res.reply("Title format is invalid")
+              }
+              else {
+                createMilestone(title, myDate)
+                res.reply("Milestones successfully created")
+              }
+              break;
+            case 5:
+              repos = message[4]
+              repos = repos.split('/')
+              title = message[3]
+              if(title == "Icebox" || title == "Backlog") {
+                createMilestone(title,myDate, repos[0], repos[1])
+                res.reply("Milestone successfully created at 'https://github.com/"+repos[0]+"/"+repos[1]+"/milestones")
+              }
+              break;
+            case 4:
+              title = message[3]
+              if(title == "Icebox" || title == "Backlog") {
+                createMilestone(title,myDate)
+                res.reply("Milestones successfully created")
+              }
+              break;
           }
         }
     })
     
- 
+    robot.hear(/bot close milestone/i, (res) => {
+      let message = res.message.text
+      message = message.split(' ')
+      if(message[0] === "bot" && message[1] === "close" && message[2] === "milestone") {
+        switch(message.length) {
+          case 7:
```

Comment:
same here, I should be able to close any milestone. And if there is no milestone I provided, it should just prompt that there is no such milestone

Reply:
@zelima imagine, bot hears message, and it is difficult to process commands, that is why we need to follow some rules. If you say `bot close milestone test test test test test` it will change entire logic.